pkg/operator/ceph/controller/spec.go

```diff
@@ -167,15 +167,14 @@ func ConfGeneratedInPodVolumeAndMount() (v1.Volume, v1.VolumeMount) {
 
 // PodVolumes fills in the volumes parameter with the common list of Kubernetes volumes for use in Ceph pods.
 // This function is only used for OSDs.
-func PodVolumes(dataPaths *config.DataPathMap, dataDirHostPath string, confGeneratedInPod bool) []v1.Volume {
+func PodVolumes(spec *cephv1.ClusterSpec, dataPaths *config.DataPathMap, dataDirHostPath string, confGeneratedInPod bool) []v1.Volume {
```

Comment:
Let's just pass the two separate host path parameters with names so we can understand why they are needed. Passing in the whole cluster spec isn't needed for now. More commonly I would say passing the spec is the right thing, but in this case I think it would cause confusion if we pass both the spec and the dataDirHostPath.
```suggestion
func PodVolumes(dataPaths *config.DataPathMap, dataHostPath, exporterHostPath string, confGeneratedInPod bool) []v1.Volume {
```

Reply:
Yeah makes sense to just take path only as input. I passed it just to make sure we're using correct dataDirHostPath from ceph cluster spec directly.